VladislavPanchehin/HW5/5.3.py

```diff
@@ -0,0 +1,21 @@
+﻿
+# variant 1
+# num = input("Entered your number: ")
+# factorial = 1
+
+
+# while 1 < num:
+#  factorial *= num
+#  num -= 1
+# print(factorial)
+
```

Comment:
good job
but look at this implementation

n = int(input("Enter your number: "))


f = 1

# Checking if the entered value is an integer and non-negative
if isinstance(n, int) and n >= 0:
    # Calculating and printing factorials
    for item in range(2, n + 1):
        f *= item
        print(f"{item}! = {f}")

else:
    # Providing feedback if the entered number is not a non-negative integer
    print("Invalid input. Please enter a non-negative integer.")

Reply:
+